internal/catalog/internal/types/health_checks_test.go

```diff
@@ -0,0 +1,245 @@
+				Prefixes: []string{""},
+			},
+			HealthChecks: []*pbcatalog.HealthCheck{
+				{
+					Name: name,
+					Definition: &pbcatalog.HealthCheck_Tcp{
+						Tcp: &pbcatalog.TCPCheck{
+							Address: "198.18.0.1",
+						},
+					},
+					Interval: durationpb.New(30 * time.Second),
+					Timeout:  durationpb.New(15 * time.Second),
+				},
+			},
+		}
+	}
+
+	type testCase struct {
+		name        string
+		err         bool
+		expectedErr utils.ErrInvalidField
+	}
+
+	// These checks are not exhaustive of the classes of names which
+	// would be accepted or rejected. The tests for the isValidDNSLabel
+	// function have more thorough testing. Here we just ensure that
+	// we can get the errNotDNSLabel error to indicate that calling
+	// that function returned false and was emitted by ValidateHealthChecks
+	cases := map[string]testCase{
+		"basic": {
+			name: "foo-check",
+		},
+		"missing-name": {
+			err: true,
+			expectedErr: utils.ErrInvalidField{
+				Name:    "name",
+				Wrapped: utils.ErrEmpty,
+			},
+		},
+		"invalid-dns-label": {
+			name: "UPPERCASE",
+			err:  true,
+			expectedErr: utils.ErrInvalidField{
+				Name:    "name",
+				Wrapped: errNotDNSLabel,
+			},
+		},
+	}
+
+	for name, tcase := range cases {
+		t.Run(name, func(t *testing.T) {
+			data := genData(tcase.name)
+			err := ValidateHealthChecks(createHealthChecksResource(t, data))
+
+			if tcase.err {
+				require.Error(t, err)
+
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestValidateHealthChecks_MissingDefinition(t *testing.T) {
+	data := &pbcatalog.HealthChecks{
+		Workloads: &pbcatalog.WorkloadSelector{
+			Prefixes: []string{""},
+		},
+		HealthChecks: []*pbcatalog.HealthCheck{
+			{
+				Name:     "test-check",
+				Interval: durationpb.New(30 * time.Second),
+				Timeout:  durationpb.New(15 * time.Second),
+			},
+		},
+	}
+
+	res := createHealthChecksResource(t, data)
+
+	err := ValidateHealthChecks(res)
+	require.Error(t, err)
+	expected := utils.ErrInvalidField{
+		Name:    "definition",
+		Wrapped: utils.ErrMissing,
+	}
+	var actual utils.ErrInvalidField
+	require.True(t, errors.As(err, &actual))
+	require.Equal(t, expected, actual)
+}
+
+func TestValidateHealthChecks_EmptySelector(t *testing.T) {
+	data := &pbcatalog.HealthChecks{
+		HealthChecks: []*pbcatalog.HealthCheck{
+			{
+				Name: "test-check",
+				Definition: &pbcatalog.HealthCheck_Tcp{
+					Tcp: &pbcatalog.TCPCheck{
+						Address: "198.18.0.1",
+					},
+				},
+				Interval: durationpb.New(30 * time.Second),
+				Timeout:  durationpb.New(15 * time.Second),
+			},
+		},
+	}
+
+	res := createHealthChecksResource(t, data)
+
+	err := ValidateHealthChecks(res)
+	require.Error(t, err)
+	expected := utils.ErrInvalidField{
+		Name:    "workloads",
+		Wrapped: utils.ErrEmpty,
+	}
+	var actual utils.ErrInvalidField
+	require.True(t, errors.As(err, &actual))
+	require.Equal(t, expected, actual)
+}
+
+func TestValidateHealthChecks_SelectorEmptyName(t *testing.T) {
```

Comment:
I think you decided to remove this type of test in the individual resources in favor of the validators_test.go for selectors

Reply:
Good catch. This has been removed.